core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -743,7 +743,15 @@ private[spark] object Utils extends Logging {
         val is = Channels.newInputStream(source)
         downloadFile(url, is, targetFile, fileOverwrite)
       case "http" | "https" | "ftp" =>
-        val uc = new URL(url).openConnection()
+        val url_object = new URL(url)
+        val uc = url_object.openConnection().asInstanceOf[HttpURLConnection]
```

Comment:
Will this work in the case of an FTP URI?
Maybe needs some pattern matching to only set this if this is an HttpURLConnection and user info is not null.